lasagna-master/ena/lasagna_master.go

```diff
@@ -0,0 +1,38 @@
+package lasagna
+
+func PreparationTime(layers []string, timePerLayer int) int {
+	const averageTimePerLayer = 2
+	if timePerLayer == 0 {
+		timePerLayer = averageTimePerLayer
+	}
+	return len(layers) * timePerLayer
+}
+
+func Quantities(layers []string) (int, float64) {
+	const noodlesPerLayer = 50 // grams
+	const saucePerLayer = 0.2  // liters
+	noodlesLayers := 0
+	sauceLayers := 0
+	for _, entry := range layers {
+		if entry == "sauce" {
+			sauceLayers++
+		} else if entry == "noodles" {
+			noodlesLayers++
+		}
+	}
+	return noodlesLayers * noodlesPerLayer, float64(sauceLayers) * saucePerLayer
+}
+
+func AddSecretIngredient(friendsList, myList []string) []string {
+	ret := myList
+	magicIngredient := friendsList[len(friendsList)-1]
+	return append(ret, magicIngredient)
+}
+
+func ScaleRecipe(quantities []float64, scaleFactor int) []float64 {
+	var scaledQuantities []float64
+	for _, entry := range quantities {
+		scaledQuantities = append(scaledQuantities, (entry/2.0)*float64(scaleFactor))
```

Comment:
You can extract `float64(scaleFactor) / 2.0` out of the loop as it's constant.

Reply:
You are right. It comes at the cost of another variable though and in this case it's probably more expensive. I tried to benchmark, but I guess the test are missing that benchmark functionality. Anyway, in principle you are right.